docs/development/pigs/pig-010.rst

```diff
@@ -0,0 +1,304 @@
+https://github.com/gammapy/gammapy/blob/v0.7/gammapy/image/radial_profile.py
+
+TODO: discuss what to keep and write it down here.
+
+Christoph: I'd prefer to replace most of ``gammapy.image`` with documentation.
+This is stuff users can script with a few lines, and trying to pre-script it in
+Gammapy to be useful to many users is very hard. We're not doing users or us a
+favour to keep this in Gammapy at this point.
+
+Tables
+------
+
+Currently we have as examples `gammapy.data.EventList`_,
+`gammapy.data.ObservationTable`_ and `gammapy.catalog.SourceCatalog`_ which each
+are a wrapper or subclass of an `astropy.table.Table`. 
+`select_sky_box`_
+`select_sky_circle`_
+`skycoord_from_table`_
+
+Re-exposed on EventList and ObservationTable methods
+
+Replace with ``select_region`` or remove completely?
+
+Offer two methods to make selection masks, and also to apply the selection?
+
+TODO: aligned lon / lat sky box - needs extra region class?
+
+Rotated regions
+---------------
+
+- One more the most complex pieces of code in Gammapy:
+  https://github.com/gammapy/gammapy/blob/master/gammapy/background/reflected.py
+- See https://github.com/gammapy/gammapy/pull/2089
+- What to do?
+
+Rotate for polygons was already implemented (see `astropy-regions GH 91`_),
+still there, even though it's currently unused.
+
+Region strings
+--------------
+
+TODO: describe possible support for region strings as input instead of as
+output. Will be required if we start storing regions in ``meta`` dicts of
+objects that we want to serialise, if we want a simple serialisation.
+
+Outlook
+=======
+
+This PIG only describes some short-term improvements to spatial regions.
+Long-term, we might want to develop a more general solution for regions that
+also includes non-spatial regions, e.g. spectral regions or time regions.
+
+Just to mention what we have at the moment: spectral "regions" are handled via
+``energy_bounds`` or ``(energy_min, energy_max)`` in a not very consistent way.
+We could look towards the Astropy `specutils`_ package and the
+`specutils.SpectralRegion`_ object as an example how to implement spectral
+regions. Currently there is plan to integrate ``astropy-regions`` and
+``specutils``, `astropy-regions GH 228`_ has been opened for discussion.
+
+Supporting an energy-dependent maximum field of view radius selection, or a
+field-of-view dependent safe energy offset might be a motivation to couple
+spatial and spectral region, possibly supported via ``gammapy.maps``.
+
+For time regions we currently often have ``(time_min, time_max)`` in the API,
+and we have the `gammapy.data.GTI`_ class to represent multiple intervals. It's
+not clear if this needs to be changed or wrapped somehow into a "time region",
+or if there are any use cases to couple time with the spatial and spectral
+regions, e.g. for a time-dependent safe energy threshold or user analysis
+option.
+
+These thoughts are related to the question in `GH 1805`_: "Introduce RegionGeom
```

Comment:
@adonath  -  Do you still think we need a `RegionGeom` or `RegionNDMap` class?

Indeed the question whether to have a map-based region subclass or not is an important one.

Could you please comment here or in #1805  how it would work? Currently I don't understand the proposal there yet.

What to do for v1.0 and for this PIG?

Reply:
Yes, I still think introducing a `RegionGeom` and `RegionNDMap` to represent spectra (or maybe lightcurves, or time dependent spectra) is a good idea. It will basically replace the current `CountsSpectrum` class, but in addition it keeps the information about the region it was extracted from in a `RegionGeom` object. This has the advantage of a uniform API with the `gammapy.maps` data structure. I.e. energy or time will be represented by `MapAxis` objects. Even if the `RegionGeom` wouldn't have a WCS object most of the methods can be defined in a meaningful way for the `RegionGeom` object as well: `RegionGeom.contains()`, `RegionGeom.center_skydir()`, `RegionGeom.downsample(axis="")`, `RegionGeom.upsample(axis="")`, `RegionGeom.get_coord()` etc.

A `WcsNDMap` could be transformed to a `RegionNDMap` by specifying a region and summing up or averaging over the spatial pixels.

If you think it fits in this PIG I'd by happy to contribute a section about the introduction of `RegionGeom` and `RegionNDMap` objects. Or maybe @AtreyeeS would you be interested in working on this?

Reply:
Discussed offline with @cdeil: we'll leave the discussion of the `RegionGeom` and `RegionNDMap` out of this PIG.

Reply:
Well, some information on regions in 1D spectral analysis will remain in the PIG, and I want to preserve the information gathered in the discussion of this PIG.
So what I plan to put is roughly this:
- at the moment we store some info about on and off regions in `meta` (have to look at where / what we store exactly) - probably it's limited to circular on region info or nothing is stored in the FITS output files after spectrum extraction.
- in the future if regions are serialisable, this info could be stored as region objects (that's what we do in HESS & HAP), which would then be a good solution IMO
- one sentence saying roughly "there is an idea to use Map objects for 1D spectra (see #1805) - it's not clear how this would related to regions (composition or inheritance) and that proposal is left as future work, not worked out in this PIG".

@adonath @registerrier - Thank you for all the feedback. I will update the PIG next week and then post it for comments for review officially, with a 1-2 week deadline for comments.

Reply:
left this out of the PIG, with only a reference to this as future work.